src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -256,4 +256,36 @@ void convertAVFrameToDecodedOutputOnCuda(
           << " took: " << duration.count() << "us" << std::endl;
 }
 
+// inspired by https://github.com/FFmpeg/FFmpeg/commit/ad67ea9
+void forceCudaCodec(
```

Comment:
Let's change the semantics of this function so that it returns the codec, if found. Then the caller is responsible for doing the assignment. That would make this signature:
```
std::optional<AVCodecPtr> findCudaCodec(
  const torch::Device& decive,
  const AVCodecID& codecId);
```
Then, inside the function, when we find the right codec, we just return it. If we loop through all available codecs and never find it, we return `std::nullopt`.